IdentityCore/src/webview/operations/MSIDWebResponseBrokerInstallOperation.m

```diff
@@ -0,0 +1,104 @@
+#import "MSIDInteractiveTokenRequestParameters.h"
+#import "MSIDWebWPJResponse.h"
+#import "MSIDTokenRequestProviding.h"
+#import "MSIDKeychainTokenCache.h"
+#import "MSIDLegacyTokenCacheAccessor.h"
+#import "MSIDDefaultTokenCacheAccessor.h"
+#import "MSIDAADV2Oauth2Factory.h"
+#import "MSIDDefaultTokenResponseValidator.h"
+#import "MSIDAccountMetadataCacheAccessor.h"
+
+@interface MSIDWebResponseBrokerInstallOperation()
+
+@property (nonatomic) NSURL *appInstallLink;
+
+@end
+
+@implementation MSIDWebResponseBrokerInstallOperation
+
+- (nullable instancetype)initWithResponse:(nonnull MSIDWebviewResponse *)response
+                                    error:(NSError * _Nullable *)error
+{
+    #if TARGET_OS_IPHONE
+        self = [super initWithResponse:response
+                                 error:error];
+        if (self)
+        {
+            if (![response isKindOfClass:MSIDWebWPJResponse.class] || [NSString msidIsStringNilOrBlank:[(MSIDWebWPJResponse *)response appInstallLink]])
+            {
+                MSID_LOG_WITH_CTX(MSIDLogLevelError, nil, @"response is not valid");
+                *error = MSIDCreateError(MSIDErrorDomain, MSIDErrorInternal, @"Wrong type of response or response does not contain a valid app install link ", nil, nil, nil, nil, nil, YES);
+                return nil;
+            }
+            
+            MSIDWebWPJResponse *wpjResponse = (MSIDWebWPJResponse *)response;
+            _appInstallLink = [NSURL URLWithString:wpjResponse.appInstallLink];
+        }
+        
+        return self;
+    #else
+        return nil;
+    #endif
+}
+
+- (void)invokeWithInteractiveTokenRequestParameters:(nonnull MSIDInteractiveRequestParameters *)interactiveTokenRequestParameters
+                               tokenRequestProvider:(nonnull id<MSIDTokenRequestProviding>)tokenRequestProvider
+                                         completion:(nonnull MSIDRequestCompletionBlock)completion
+{
+    #if TARGET_OS_IPHONE
+        if ([interactiveTokenRequestParameters isKindOfClass:MSIDInteractiveTokenRequestParameters.class])
+        {
+            NSError *brokerError;
+            MSIDBrokerInteractiveController *brokerController = [[MSIDBrokerInteractiveController alloc] initWithInteractiveRequestParameters:(MSIDInteractiveTokenRequestParameters *)interactiveTokenRequestParameters
+                                                                                                                         tokenRequestProvider:tokenRequestProvider
+                                                                                                                            brokerInstallLink:self.appInstallLink
+                                                                                                                                        error:&brokerError];
+            if (brokerError)
+            {
+                MSID_LOG_WITH_CTX(MSIDLogLevelError, nil, @"Not able to create brokerController");
```

Comment:
can we also log the actual error for investigation?

Reply:
Updated